File: 11vaFeriaAcademicayCultural/Basico/Problema3/problema3.cpp
 /*Problema 3 Validando Password
Propuesta de solucion
autor: Luis Arturo De La Garza Navel
C++*/
#include <iostream>
#include<string> //libreria para manejo de la clase string
#include<ctype.h> // libreria para la funcion que verifica si un caracter es digito
using namespace std;

int main() {
  string password; // string para leer la entrada
  bool password_valido = false; // bandera booleana para validar password, inicia en false
  cin>>password;
  //validamos si la cadena no inicie con digito y la longitud este en el limite definido
  if(!isdigit(password[0]) && 5 <= password.length() &&  password.length() <= 13){
    for(int i = 1; i < password.length(); i++){
      if(isdigit(password[i])){
        //si encontramos un digito dejamos de iterar y validamos el password
        password_valido = true;
        break;
      }
    }
  if(password_valido){
    cout<<"Password valido"<<endl;
  }
  else{
    cout<<"Password invalido"<<endl;
  }

  return 0;
}

File: 11vaFeriaAcademicayCultural/Basico/Problema3/problema3.py
"""Problema 3 Validando Password
Propuesta de solucion
autor: Luis Arturo De La Garza Navel
Python 2.7"""

"""el uso del modulo sys para entrada/salida de datos es altamente
recomendado en programacion competitiva (ojo: no todas las competencias
permiten su uso), pero puede usar los metodos propios definidos en
python (raw_input y print)"""

from sys import stdin, stdout

if __name__ == '__main__':
    #  el metodo readline de stdin lee la siguiente linea de la entrada
    #  incluyendo el salto de linea, para eliminarlo usamos el metodo strip (sin parametros)
    #  de la clase str para eliminar los caracteres no deseados (espacios y saltos de linea)
    password = stdin.readline().strip()
    password_valido = False #  bandera booleana para validar password, inicia en False
    #  Validamos que password no inicie con un digito para ser valido
    #  ademas que su longitud sea entre 5 a 13
    if not password[0].isdigit() and 5 <= len(password) <= 13 :
        for caracter in password[1:]:
            if caracter.isdigit():
                #  si encontramos un digito dejamos de iterar y validamos el password
                password_valido = True
                break

    if password_valido:
        stdout.write('Password valido\n')
    else:
        stdout.write('Password invalido\n')
